db/mongodb.go

```diff
@@ -247,3 +258,45 @@ func (m *MongoDB) CreateExtraIndexes(idxs []string) error {
 	slog.Info(fmt.Sprintf("%d index(es) successfully created in the collection %s", len(r), companyTableName))
 	return nil
 }
+
+func (m *MongoDB) Search(q Query) (string, error) {
+	uf := strings.ToUpper(q.Uf)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	coll := m.db.Collection(companyTableName)
+	f := bson.M{"json.uf": uf}
+	if q.Cursor != "" {
+		f["json.cnpj"] = bson.M{"$gt": q.Cursor}
+	}
+	opts := options.Find().
+		SetSort(bson.D{{Key: "json.cnpj", Value: 1}}).
+		SetLimit(q.Limit)
+	c, err := coll.Find(ctx, f, opts)
+	if err != nil {
+		return "", fmt.Errorf("error running query %#v: %w", uf, err)
+	}
+	defer c.Close(ctx)
+
+	var results []record
+	if err := c.All(ctx, &results); err != nil {
+		return "", fmt.Errorf("error decoding results: %w", err)
+	}
+
+	if len(results) == 0 {
+		return "", nil
+	}
+
+	var cs []transform.Company
+	var cur string
+	for _, r := range results {
+		cs = append(cs, r.Json)
+		cur = r.Json.CNPJ
```

Comment:
Lembra que pedi [para não (re)escrevermos `cur` mil vezes](https://github.com/cuducos/minha-receita/pull/346#discussion_r2207378265). Você _não_ resolve isso colocando essa linha no `for` _loop_.

```suggestion
```